src/util.ml

```diff
@@ -243,13 +249,14 @@ value gen_decline2 wt s1 s2 =
             | None -> (":", i) ]
         | '[' ->
             try
+              let k = String.index_from wt i '|' in
```

Comment:
Here my point is not that `k` is a wrong name. It's because you introduce a new delimiter `|` for the translation.